aea/skills/base.py

```diff
@@ -389,18 +389,70 @@ def parse_module(
 class Task(SkillComponent):
     """This class implements an abstract task."""
 
-    def __init__(self, *args, **kwargs):
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        skill_context: Optional[SkillContext] = None,
```

Comment:
both `name` and `skill_context` are obsolete - especially the `skill_context`, since it is not necessary for all the use cases. 

At the end of the day, the user must be able to feed the data he needs to the `execute` function.

Also, `skill_context` is not pickable (due to the queue probably) - which prevents the `multiprocessing.Pool` working properly...

Reply:
Ok so then why do you add it? Remove it :)